### ISSUE TITLE
Feature Request: Make VR Optional

### ISSUE THREAD

**Issue by mhughson** [N7Alpha/UnrealLibretro#31]
It would be helpful if the VR dependencies of this plug-in were _optional_. There are platforms where VR does not exist, and this plug-in will not compile there.

If it's easier, a "no-vr" branch would work too.

**Comment by N7Alpha** [comment]
Yeah I think that's true. I've been working on netplay stuff right now, and I don't really have a timeline on when I could get this done. I think this is worth doing though. Also which platform are you trying to build for? That way I could test it.

I'm not quite sure what the best approach to accomplish this is at the moment though. I think I should be able to just fence the VR specific stuff into an `UnrealLibretroVR` code module. I might still have problems with blueprints though since I can't move them into the code module. I can maybe flag them to not package in one of the `Build.cs` files? I've struggled with the Unreal Engine build system in the past, and I could hit a roadblock with this. Alternatively, I might change the structure of the project such that it is an example project and move the VR code into the `Content` and `Source` folders for the project and keep the plugin more or less purely the `LibretroCoreInstance`. That has a further set of advantages since maintaining engine backwards compatibility is easier that way. Also people have suggested having an example project would be good since that means potentially less setup voodoo required.

**Comment by mhughson** [comment]
I am looking a home consoles. Not sure if you have access to that.

I think seperating out "core", "VR" and "examples" into separate plugins would be clean. That way, most users would include "core" and maybe "VR" and all the example pawn/character etc stuff could be isolated to example implementations (which maybe some people want, but folks building their own games probably already have pawns etc).

**Comment by mhughson** [comment]
I'm honestly not even sure if the emulator cores are going to work on consoles.

**Comment by N7Alpha** [comment]
> I think seperating out "core", "VR" and "examples" into separate plugins would be clean. That way, most users would include "core" and maybe "VR" and all the example pawn/character etc stuff could be isolated to example implementations (which maybe some people want, but folks building their own games probably already have pawns etc).

I'll attempt to do it as a [Unreal Engine Module](https://docs.unrealengine.com/4.27/en-US/ProgrammingAndScripting/ProgrammingWithCPP/Modules/). A blunter tool could be required though if I can't get that working though.  I was also just verifying my own API's were reasonable and not too cumbersome.

> I'm honestly not even sure if the emulator cores are going to work on consoles.

Yeah you should check if they support runtime dll loading. Some OS's consider this a security risk and disable it. Also check for OpenGL support (if you want hardware accelerated cores). There are workarounds for this static linking and ANGLE, but that requires doing considerable work. Also I don't think many console manufacturers like emulators. Microsoft is the only exception I think so I assume you're testing on that or something.
